### DISCUSSION TITLE
Ordering and lifetime for 2D image annotations

### DISCUSSION THREAD

**Discussion by vivek-rane** [foxglove/community#635] [Q&A]
When I have multiple detections (say rectangles) at the same location, I want to display them in a certain order so that I can either:
1. Put the most important one "on top" (drawn last)
2. Make the ones "at the bottom" (drawn first) thicker so they can still be seen.

This is, however, only possible if I can enforce the ordering between ImageMarkers. If there is some way to add this ordering, it would really be helpful.

I can think of a couple of ways of doing this - either should work:
1. A straight up drag and drop for the markers listed in an image panel settings
2. A "weight" field for each `ImageMarkers` object with the markers with more weight being at the "bottom" (drawn first)

A second problem with these markers is that even though the detections are per frame, the markers last for way too long. It would be nice/consistent to have the equivalent of `lifetime` that the 3D markers have in the 2D world as well.

**Comment by MaiTarekAlazab** [comment]
I had the same issue with the image marker lifetime, it is like the lifetime can't be controlled whether increased or decreased, it keeps showing until new detection comes, if not it lasts forever.
